spec/services/api/beta/search_service_spec.rb

```diff
@@ -103,6 +103,31 @@
     it_behaves_like 'a redirecting search result', '0101210000-80' # Subheading
     it_behaves_like 'a redirecting search result', '0101210000380' # Heading
     it_behaves_like 'a redirecting search result', '010121000038123' # Heading
+
+    context 'when the search query has multiple corresponding search references' do
+      let(:search_query) { 'same' }
+
+      before do
+        create(:heading, :with_search_reference, title: 'same')
+        create(:commodity, :with_search_reference, title: 'same')
+
+        search_result = Hashie::TariffMash.new(
+          JSON.parse(
+            file_fixture('beta/search/goods_nomenclatures/single_hit.json').read,
+          ),
+        )
+        allow(TradeTariffBackend.v2_search_client).to receive(:search).and_return(search_result)
+        allow(Api::Beta::SearchQueryParserService).to receive(:new).and_return(
+          instance_double('Api::Beta::SearchQueryParserService', call: build(:search_query_parser_result)),
+        )
+        allow(Beta::Search::GoodsNomenclatureQuery).to receive(:build).and_return(build(:goods_nomenclature_query))
+        allow(Beta::Search::OpenSearchResult::WithHits).to receive(:build).and_call_original
+
+        call
```

Comment:
After some initial confusion this is not a `call` on the current object but a named subject. I feel `call` is sufficiently generic confuse other developers

Reply:
I reflected on this quite a bit and looked at other services and decided I agree. This is also not the first time I've had this conversation so yeah